provider/pkg/await/await.go

```diff
@@ -438,13 +444,29 @@ func Update(c UpdateConfig) (*unstructured.Unstructured, error) {
 			// Issue patch request.
 			// NOTE: We can use the same client because if the `kind` changes, this will cause
 			// a replace (i.e., destroy and create).
-			currentOutputs, err = client.Patch(c.Context, c.Inputs.GetName(), types.ApplyPatchType, objYAML, options)
-			if err != nil {
-				if errors.IsConflict(err) {
-					err = fmt.Errorf("Server-Side Apply field conflict detected. see %s for troubleshooting help\n: %w",
-						ssaConflictDocLink, err)
+			maybeRetry := true
+			for {
```

Comment:
Logic here mostly looks good to me.
nit: Is it possible to functionalize this for loop instead, perhaps into a recursive like function since we have many nested code blocks now that might make it harder for us to read.

Reply:
We can iterate on this in a follow-up PR.